js/url_search_params.ts

```diff
@@ -119,92 +121,29 @@ export class URLSearchParams {
    *       searchParams.sort();
    */
   sort(): void {
-    this.params = this.params.sort(
+    this[params] = this[params].sort(
       (a, b) => (a[0] === b[0] ? 0 : a[0] > b[0] ? 1 : -1)
     );
   }
 
-  /** Calls a function for each element contained in this object in
-   * place and return undefined. Optionally accepts an object to use
-   * as this when executing callback as second argument.
-   *
-   *       searchParams.forEach((value, key, parent) => {
-   *         console.log(value, key, parent);
-   *       });
-   *
-   */
-  forEach(
-    callbackfn: (value: string, key: string, parent: URLSearchParams) => void,
-    // tslint:disable-next-line:no-any
-    thisArg?: any
-  ) {
-    if (typeof thisArg !== "undefined") {
-      callbackfn = callbackfn.bind(thisArg);
-    }
-    for (const [key, value] of this.entries()) {
-      callbackfn(value, key, this);
-    }
-  }
-
-  /** Returns an iterator allowing to go through all keys contained
-   * in this object.
-   *
-   *       for (const key of searchParams.keys()) {
-   *         console.log(key);
-   *       }
-   */
-  keys(): IterableIterator<string> {
-    const list = this.params.map(param => param[0]);
-    const iterators = list.values();
-    return new CreateIterableIterator(iterators);
-  }
-
-  /** Returns an iterator allowing to go through all values contained
-   * in this object.
-   *
-   *       for (const value of searchParams.values()) {
-   *         console.log(value);
-   *       }
-   */
-  values(): IterableIterator<string> {
-    const list = this.params.map(param => param[1]);
-    const iterators = list.values();
-    return new CreateIterableIterator(iterators);
-  }
-
-  /** Returns an iterator allowing to go through all key/value
-   * pairs contained in this object.
-   *
-   *       for (const [key, value] of searchParams.entries()) {
-   *         console.log(key, value);
-   *       }
-   */
-  entries(): IterableIterator<[string, string]> {
-    const iterators = this.params.values();
-    return new CreateIterableIterator(iterators);
-  }
-
-  /** Returns an iterator allowing to go through all key/value
-   * pairs contained in this object.
-   *
-   *       for (const [key, value] of searchParams[Symbol.iterator]()) {
-   *         console.log(key, value);
-   *       }
-   */
-  [Symbol.iterator](): IterableIterator<[string, string]> {
-    return this.entries();
-  }
-
   /** Returns a query string suitable for use in a URL.
    *
    *        searchParams.toString();
    */
   toString(): string {
-    return this.params
+    return this[params]
       .map(
         tuple =>
           `${encodeURIComponent(tuple[0])}=${encodeURIComponent(tuple[1])}`
       )
       .join("&");
   }
 }
+
+// @internal
+// tslint:disable:max-line-length
+// tslint:disable-next-line:variable-name
```

Comment:
```suggestion
// tslint:disable-next-line:variable-name max-line-length
```

But even then, Prettier should break the generic arguments up.

Reply:
Thanks.

Reply:
So yeah, now with the wrapping after running through prettier, `max-line-length` is redundant.